scripts/add_electricity.py

```diff
@@ -338,6 +348,24 @@ def attach_wind_and_solar(
                 p_max_pu=ds["profile"].transpose("time", "bus").to_pandas(),
             )
 
+        if suptech == "offwind":
+            continue
+            # df.carrier.mask(df.technology=="Offshore", "offwind-ac",inplace=True)
+
+        df.carrier.mask(df.technology == "Onshore", "onwind", inplace=True)
+
+        gens = n.generators[lambda df: df.carrier == tech]
+        buses = n.buses.loc[gens.bus.unique()]
+        gens_per_bus = gens.groupby("bus").p_nom.count()
+
+        if not gens.empty and not df.query("carrier == @tech").empty:
+            caps = map_country_bus(df.query("carrier == @tech"), buses)
+            caps = caps.groupby(["bus"]).p_nom.sum()
+            caps = caps / gens_per_bus.reindex(caps.index, fill_value=1)
+
+            n.generators.p_nom.update(gens.bus.map(caps).dropna())
```

Comment:
why updating the capacities and not add already above at `n.madd()`?

Reply:
@pz-max, we can do that. For that we need to move all added code after condition with hydro: 
![image](https://github.com/pypsa-meets-earth/pypsa-earth/assets/113768325/82e5ff8f-b374-4c4e-979b-4b87252f05f3)

Reply:
Let's do that :+1: 
Can you make these changes here?